src/Umbraco.Core/Telemetry/TelemetryService.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Options;
+using Umbraco.Cms.Core.Configuration;
+using Umbraco.Cms.Core.Configuration.Models;
+using Umbraco.Cms.Core.Manifest;
+using Umbraco.Cms.Core.Telemetry.Models;
+using Umbraco.Extensions;
+
+namespace Umbraco.Cms.Core.Telemetry
+{
+    /// <summary>
+    /// Service which gathers the data for telemetry reporting
+    /// </summary>
+    public sealed class TelemetryService
+    {
+        private readonly GlobalSettings _globalSettings;
+        private readonly IManifestParser _manifestParser;
+        private readonly IUmbracoVersion _umbracoVersion;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TelemetryService"/> class.
+        /// </summary>
+        public TelemetryService(
+            IOptions<GlobalSettings> globalSettings,
+            IManifestParser manifestParser,
+            IUmbracoVersion umbracoVersion)
+        {
+            _globalSettings = globalSettings.Value;
```

Comment:
The _globalSettings.Id is set runtime if missing, so I guess we need to add this as IOptionsMonitor and update the cached value `OnChange`